aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -149,15 +149,15 @@ void cpu_max_pool_channels_last(
 
       // Pass I: init out lane
       iVec index0_vec = iVec(ih0 * input_width + iw0);
-      Vec out_vec = Vec(-std::numeric_limits<scalar_t>::infinity());
+      Vec out_vec = Vec(input.is_quantized() ? std::numeric_limits<scalar_t>::lowest() : -std::numeric_limits<scalar_t>::infinity());
```

Comment:
I'm afraid the test isn't going to get optimized away. Do you have benchmark numbers here? You could also make the `is_quantized` a boolean template parameter, so it is guaranteed to be statically known.